internal/data_source_cedar_schema.go

```diff
@@ -0,0 +1,84 @@
+package provider
+
+import (
+	"context"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+	"terraform-provider-boxer/pkg/generated/api"
+
+	"github.com/hashicorp/terraform-plugin-framework/datasource"
+	"github.com/hashicorp/terraform-plugin-framework/datasource/schema"
+)
+
+// Ensure the implementation satisfies the expected interfaces.
+var (
+	_ datasource.DataSource              = &cedarSchemaDataSource{}
+	_ datasource.DataSourceWithConfigure = &cedarSchemaDataSource{}
+)
+
+// CedarSchemaDataSource is a helper function to simplify the provider implementation.
+func CedarSchemaDataSource() datasource.DataSource {
```

Comment:
```suggestion
func CedarSchemaDataSource() *datasource.DataSource {
```
since you return a pointer?